### ISSUE TITLE
Better error message / restrict audit to body of #ember-testing

### ISSUE THREAD

**Issue by Gaurav0** [ember-a11y/ember-a11y-testing#148]
When running an integration test on a component that was not focusable, I receive this error message:

```
Assertion Failed:
The page should have no accessibility violations. Violations:
[moderate]: Ensure that scrollable region has keyboard access 
Violated 1 time. Offending nodes are: 
<div id="ember-testing-container" style="visibility: visible; position: relative;">
https://dequeuniversity.com/rules/axe/3.5/scrollable-region-focusable?application=axeAPI
    at Object.assert (http://localhost:7357/assets/vendor.js:37196:15)
    at a11yAuditCallback (http://localhost:7357/assets/test-support.js:14250:34)
    at invokeCallback (http://localhost:7357/assets/vendor.js:61742:17)
    at publish (http://localhost:7357/assets/vendor.js:61725:9)
    at http://localhost:7357/assets/vendor.js:56152:9
    at invoke (http://localhost:7357/assets/vendor.js:54358:16)
    at Queue.flush (http://localhost:7357/assets/vendor.js:54249:13)
    at DeferredActionQueues.flush (http://localhost:7357/assets/vendor.js:54446:21)
    at Backburner._end (http://localhost:7357/assets/vendor.js:54980:34)
    at Backburner._boundAutorunEnd (http://localhost:7357/assets/vendor.js:54649:14)
```

The error message points to the ember testing container and says to add a tabIndex, which can't be done because it is injected by ember-qunit. Either let's have the a11y audit given a document restricted to contents of #ember-testing or see if we can improve the error message.

**Comment by drewlee** [comment]
@Gaurav0, a default `a11yAudit` run is already restricted to the fixture container (https://github.com/ember-a11y/ember-a11y-testing/blob/master/addon-test-support/audit.js#L66). I suspect this may be caused by some other underlying issue or a change on the axe-core side of things. Can you provide a set of steps or a repo to reliably reproduce, along with the axe-core version? Thanks!

**Comment by scalvert** [comment]
Since we don't have a reproduction, I'm going to close this for now. @Gaurav0, feel free to reopen with a reproduction if this issue is still persisting!

**Comment by asppsa** [comment]
I am seeing this issue in some code I am working on as well.  I haven't had time yet to produce a repro in some stand-alone code, but I have found that configuring the hooks like so seems to fix the issue:

```js
setupGlobalA11yHooks(
  () => true,
  (_contextSelector, axeOptions) => a11yAudit('#ember-testing-container > *', axeOptions),
)
```

That is, I am changing the context from `#ember-testing-container` to `#ember-testing-container > *` - all elements inside the container, rather than the container itself.

https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#context-parameter says that using a CSS selector will tell axe which part of the document to be analysed.  Therefore, I'm working on the hypothesis that using `#ember-testing-container` as the selector tells it to include the container element in its analysis, and that that causes issues with certain types of layouts.

It's still possible that the real issue is something to do with how my tests are setup - when I get some time I'll try to make a plain Ember repo that reproduces the issue to confirm.

**Comment by fdeters** [comment]
Commenting because I regularly run into this problem and always retrace my steps to this issue. Here's what seems to work for me (very similar to the previous comment but with async):

```js
setupGlobalA11yHooks(
  () => true,
  async (contextSelector, axeOptions) => {
    await a11yAudit('#ember-testing-container > *', axeOptions);
  },
);
```
